File: springboot2/src/main/java/com/demo/model/SelectOption.java
package com.demo.model;

import com.demo.annotation.BQAQuery;
import lombok.Data;
import lombok.EqualsAndHashCode;
import org.springframework.data.mongodb.core.mapping.Document;

import java.util.List;

@Data
@EqualsAndHashCode(callSuper = true)
@Document
public class SelectOption extends BaseModel {

    @BQAQuery
    private String name;
    @BQAQuery
    private String module;
    private List<Option> options;

    @Data
    public static class Option {
        private String value;
        private String label;
        private Integer sortOrder;
    }
}

File: vite-project-react18/src/models/selectOption.ts
interface IOption {
  value: string;
  label: string;
  sortOrder: number;
}

interface ISelectOption {
  id: string;
  createdAt: string;
  createdBy: string;
  updatedAt: string;
  updatedBy: string;
  name: string;
  module: string;
  options: Option[];
}

export type SelectOption = Partial<ISelectOption>;
export type Option = Partial<IOption>;
